### DISCUSSION TITLE
Base path and Redirection with uri!

### DISCUSSION THREAD

**Discussion by xemwebe** [rwf2/Rocket#1941] [Questions]
The following simple server does not behave as I would expect:
```rust
#[macro_use] extern crate rocket;
use rocket::response::Redirect;

#[get("/hello")]
async fn hello() -> String {
    "Hello, world!".to_string()
}

#[get("/redirect")]
async fn redirect() -> Redirect {
    Redirect::to(uri!(hello))
}

#[launch]
async fn rocket() -> _ {
    let base_path = "/base";
    let rocket_config = rocket::Config::default();
    rocket::custom(rocket_config)
        .mount(base_path, routes![
            hello,
            redirect,
        ])
}
```
If the server is started, paths `/base/hello` and `/base/redirect` are available. However, what I didn't expect is that `/base/redirect` redirects to `/hello` instead of to `/base/hello`, which makes the macro `uri!` useless except if `base_path` is empty.

My question is: Is this expected, or is this a bug? Or did I do something wrong? 

What would be the most efficient way to put all paths of the web app behind some base path but still use the convenience macro `uri!`?

**Comment by daaku** [comment]
This is one way to handle this: https://github.com/SergioBenitez/Rocket/blob/master/examples/cookies/src/session.rs#L32
